### ISSUE TITLE
Drag Gesture Issue

### ISSUE THREAD

**Issue by devmlew** [Mijick/NavigationView#46]
Hello, I am seeing odd gesture behavior when navigating more than 1 view deep. I am also seeing this behavior with rotation. I have attached some GIFs and a test snippet to recreate the issue.

![Simulator Screen Recording - iPhone 15 Pro - 2024-07-07 at 02 20 09](https://github.com/Mijick/NavigationView/assets/11561943/c8dc95f6-7b06-48f9-a639-139f3ac00242)

![Simulator Screen Recording - iPhone 15 Pro - 2024-07-07 at 02 21 36](https://github.com/Mijick/NavigationView/assets/11561943/79349f1c-7a10-41de-97f6-b777de06feef)

```
import SwiftUI
import MijickNavigationView

var navigationConfig: NavigationGlobalConfig {
    var config = NavigationGlobalConfig()
    config.backgroundColour = Color(.gray)
    config.backGestureThreshold = 0.25
    config.backGesturePosition = .anywhere
    return config
}

@main
struct naviIssueApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
               .implementNavigationView(config: navigationConfig)
        }
    }
}

struct ContentView: NavigatableView {
    func configure(view: NavigationConfig) -> NavigationConfig {
        view
            .backgroundColour(.purple)
            .navigationBackGesture(.drag)
            .ignoresSafeArea(.all, .bottom)
    }
    var body: some View {
        VStack(spacing: 0) {
            Text("ContentView1")
            Spacer()
            
            Button(action: {
                ContentView2().push(with: .horizontalSlide)
            }) {
                Text("Jump to ContentView2")
                    .foregroundColor(.black)
            }
            
            Spacer()
        }
    }
}

struct ContentView2: NavigatableView {
    func configure(view: NavigationConfig) -> NavigationConfig {
        view
            .backgroundColour(.blue)
            .navigationBackGesture(.drag)
            .ignoresSafeArea(.all, .bottom)
    }
    var body: some View {
        VStack(spacing: 0) {
            Text("ContentView2")
            Spacer()
            
            Button(action: pop) { 
                Text("Pop").foregroundColor(.black)
            }
            
            Button(action: {
                ContentView3().push(with: .horizontalSlide)
            }) { 
                Text("Jump to ContentView3")
                    .foregroundColor(.black)
            }
            
            Spacer()
        }
    }
}

struct ContentView3: NavigatableView {
    func configure(view: NavigationConfig) -> NavigationConfig {
        view
            .backgroundColour(.green)
            .navigationBackGesture(.drag)
            .ignoresSafeArea(.all, .bottom)
    }
    var body: some View {
        VStack(spacing: 0) {
            Text("ContentView3")
            Spacer()
            
            Button(action: pop) { 
                Text("Pop").foregroundColor(.black)
            }
            
            Spacer()
        }
    }
}

#Preview {
    ContentView()
        .implementNavigationView(config: navigationConfig)
}
```

**Comment by FulcrumOne** [comment]
Hey @devmlew!

Thanks for your report, we'll check it out

**Comment by ByLiangCheng** [comment]
I also encountered this problem

**Comment by FulcrumOne** [comment]
Hey guys!

This problem should be fixed in `Patch-1.1.2` branch. Could you please check it? Thanks!

**Comment by ByLiangCheng** [comment]
> Hey guys!
> 
> This problem should be fixed in `Patch-1.1.2` branch. Could you please check it? Thanks!

I switched to this branch and it worked, thank you.
